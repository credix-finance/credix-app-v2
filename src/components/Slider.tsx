import React from "react";
import { Slider as AntdSlider } from "antd";
import { SliderSingleProps as AntdSliderSingleProps } from "antd/lib/slider";

interface SliderProps {
	/**
	 * The value of the slider.
	 */
	value: AntdSliderSingleProps["value"];
	/**
	 * Optional message to be displayed when the slider is "full".
	 */
	fullLabel?: string;

	/**
	 * Optional slider max value.
	 */
	max?: AntdSliderSingleProps["max"];
}

export const Slider = ({ value, fullLabel, max = 100, ...props }: SliderProps) => {
	const isFull = value >= max;

	return (
		<div className="flex flex-col">
			<div className="font-semibold">
				{isFull ? <span>{fullLabel}</span> : <span>{value}%</span>}
			</div>
			<div className="block">
				<AntdSlider
					value={value}
					max={max}
					handleStyle={isFull && { display: "none" }}
					{...props}
				/>
			</div>
		</div>
	);
};
