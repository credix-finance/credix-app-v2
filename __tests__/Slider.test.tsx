import React from "react";
import renderer from "react-test-renderer";
import { Slider } from "@components/Slider";

test("Default slider", () => {
	const component = renderer.create(<Slider value={50} />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test("Full slider", () => {
	const component = renderer.create(<Slider value={50} />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
