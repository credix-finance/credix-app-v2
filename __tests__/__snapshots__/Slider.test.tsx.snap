// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default slider 1`] = `
<div
  className="flex flex-col"
>
  <div
    className="font-semibold"
  >
    <span>
      50
      %
    </span>
  </div>
  <div
    className="block"
  >
    <div
      className="ant-slider"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="ant-slider-rail"
        style={Object {}}
      />
      <div
        className="ant-slider-track"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "width": "50%",
          }
        }
      />
      <div
        className="ant-slider-step"
      />
      <div
        aria-disabled={false}
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={50}
        className="ant-slider-handle"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="slider"
        style={
          Object {
            "left": "50%",
            "right": "auto",
            "transform": "translateX(-50%)",
          }
        }
        tabIndex={0}
      />
      <div
        className="ant-slider-mark"
      />
    </div>
  </div>
</div>
`;

exports[`Full slider 1`] = `
<div
  className="flex flex-col"
>
  <div
    className="font-semibold"
  >
    <span />
  </div>
  <div
    className="block"
  >
    <div
      className="ant-slider"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="ant-slider-rail"
        style={Object {}}
      />
      <div
        className="ant-slider-track"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "width": "100%",
          }
        }
      />
      <div
        className="ant-slider-step"
      />
      <div
        aria-disabled={false}
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={100}
        className="ant-slider-handle"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="slider"
        style={
          Object {
            "display": "none",
            "left": "100%",
            "right": "auto",
            "transform": "translateX(-50%)",
          }
        }
        tabIndex={0}
      />
      <div
        className="ant-slider-mark"
      />
    </div>
  </div>
</div>
`;
